python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -331,3 +332,144 @@ def get_units(value):
+            - year
+            - month
+            - week
+            - day
+            - hour
+            - minute
+            - second
+            - microsecond
+            - millisecond
+            - nanosecond
+        Finally, cuDF does not yet support rounding via a `normalize`
+        keyword argument.
+
+        Parameters
+        ----------
+        n : int, default 1
+            The number of time periods the offset represents.
+        **kwds
+            Temporal parameter that add to or replace the offset value.
+            Parameters that **add** to the offset (like Timedelta):
+            - months
+
+        See Also
+        --------
+        pandas.DateOffset : The equivalent Pandas object that this
+        object replicates
+
+        Examples
+        --------
+        >>> from cudf import DateOffset
+        >>> ts = cudf.Series([
+            "2000-01-01 00:00:00.012345678",
+            "2000-01-31 00:00:00.012345678",
+            "2000-02-29 00:00:00.012345678",
+        ], dtype='datetime64[ns])
+        >>> ts + DateOffset(months=3)
+        0   2000-04-01 00:00:00.012345678
+        1   2000-04-30 00:00:00.012345678
+        2   2000-05-29 00:00:00.012345678
+        dtype: datetime64[ns]
+        >>> ts - DateOffset(months=12)
+        0   1999-01-01 00:00:00.012345678
+        1   1999-01-31 00:00:00.012345678
+        2   1999-02-28 00:00:00.012345678
+        dtype: datetime64[ns]
+        """
+        if normalize:
+            raise NotImplementedError(
+                "normalize not yet supported for DateOffset"
+            )
+
+        # TODO: Pandas supports combinations
+        if len(kwds) > 1:
+            raise NotImplementedError("Multiple time units not yet supported")
+
+        all_possible_kwargs = {
+            "years",
+            "months",
+            "weeks",
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "microseconds",
+            "nanoseconds",
+            "year",
+            "month",
+            "week",
+            "day",
+            "hour",
+            "minute",
+            "second",
+            "microsecond",
+            "millisecond" "nanosecond",
+        }
+
+        supported_kwargs = {"months"}
+
+        scalars = {}
+        for k, v in kwds.items():
+            if k in all_possible_kwargs:
+                # Months must be int16
+                dtype = "int16" if k == "months" else None
+                scalars[k] = cudf.Scalar(v, dtype=dtype)
+
+        super().__init__(n=n, normalize=normalize, **kwds)
+
+        wrong_kwargs = set(kwds.keys()).difference(supported_kwargs)
+        if len(wrong_kwargs) > 0:
+            raise ValueError(
+                f"Keyword arguments '{','.join(list(wrong_kwargs))}'"
+                " are not yet supported in cuDF DateOffsets"
+            )
+        self._scalars = _DateOffsetScalars(scalars)
+
+    def _generate_column(self, size, op):
```

Comment:
Can you add a `#todo` comment with a link to the github issue for the scalar function overload?